summarize/Readme.md

```diff
@@ -39,8 +70,77 @@ $ summarize pid-{pid}
 | mir_const              | 117.82ms  | 1.081           | 1050       | 30         | 0.00ns       | 0.00ns                |
 +------------------------+-----------+-----------------+------------+------------+--------------+-----------------------+
 
-(many more rows elided)
+(rows elided)
 
 Total cpu time: 10.896488447s
+```
 
+## Profiling your own build of rustc
+
+You can also profile your own custom build of rustc. First you'll have to clone the
+[rust][rust-repo] repo and compile it. You can find the full guide on doing this
+[here][compiling-rust], but if you've never built rustc before, we suggest starting with
+
+[rust-repo]: https://github.com/rust-lang/rust
+[compiling-rust]: https://rust-lang.github.io/rustc-guide/how-to-build-and-run.html
+
+```bash
+$ git clone https://github.com/rust-lang/rust.git
+$ ./x.py build
+# This will take a while...
+$ rustup toolchain link mytoolchain build/x86_64-unknown-linux-gnu/stage2
 ```
+
+Where `mytoolchain` is the name of your custom toolchain. Now we do more or less the same
+as before: (with regex as example)
+
+```bash
+$ git clone https://github.com/rust-lang/regex.git
+$ cd regex
+$ cargo +mytoolchain rustc -- -Z self-profile
```

Comment:
Did `stage0` toolchain work for you? For me it failed with `error: the option Z is only accepted on the nightly compiler`. However, `stage1` works.

Reply:
`stage0` toolchain won't work because its compiler is the stable Rust compiler the bootstrap process downloaded at the beginning. `stage1` is the first stage that could work but procedural macros don't work in the `stage1` compiler. So for many crates, `stage1` also will not work (if they use proc macros or crates that use proc macros). `stage2` will always work. So the above instructions to link to the `stage2` compiler are correct.

Reply:
Thank you